dts/bindings/comparator/nordic,nrf-comp.yaml

```diff
@@ -0,0 +1,131 @@
+# Copyright (c) 2024 Nordic Semiconductor ASA
+# SPDX-License-Identifier: Apache-2.0
+
+description: |
+  Nordic nRF COMP (analog COMParator)
+
+  This comparator has varying configurations which require varying
+  properties be set in the devicetree.
+
+  The following example displays the minimum node layout:
+
+    comp: comp@deadbeef {
+            compatible = "nordic,nrf-comp";
+            reg = <0xdeadbeef 0x1000>;
+            interrupts = <0 NRF_DEFAULT_IRQ_PRIORITY>;
+            status = "disabled";
+    };
+
+  Enabling the comparator node requires setting the default
+  configuration of the comparator.
+
+  The following example displays enabling the comparator in
+  single-ended mode, selecting an internal reference:
+
+    &comp {
+            status = "okay";
+            positive-input = "INPUT_0";
+            reference-selection = "INT_1V2";
+            main-mode = "SE";
+            speed-mode = "NORMAL";
+            threshold-up = <36>;
+            threshold-down = <30>;
+            isource = "OFF";
+    };
+
+  To select an external reference, select the "AREF"
+  reference and add the external reference:
+
+    &comp {
+            ...
+            reference-selection = "AREF";
+            external-reference = "EXT_REF_1";
+            ...
+    };
+
+  The following example displays enabling the comparator
+  in differential mode:
+
+    &comp {
+            status = "okay";
+            positive-input = "INPUT_0";
+            external-reference = "EXT_REF_1";
+            main-mode = "DIFF";
+            speed-mode = "NORMAL";
+            hysteresis = "50MV";
+            isource = "OFF";
+    };
+
+compatible: "nordic,nrf-comp"
+
+include: base.yaml
+
+properties:
+  positive-input:
+    type: string
+    enum:
+      - "INPUT_0"
+      - "INPUT_1"
+      - "INPUT_2"
+      - "INPUT_3"
+      - "INPUT_4"
+      - "INPUT_5"
+      - "INPUT_6"
+      - "INPUT_7"
+      - "INPUT_VDD_DIV2"
+      - "INPUT_VDD_DIV5"
+
+  external-reference:
+    type: string
+    enum:
+      - "EXT_REF_0"
+      - "EXT_REF_1"
+      - "EXT_REF_2"
+      - "EXT_REF_3"
+      - "EXT_REF_4"
+      - "EXT_REF_5"
+      - "EXT_REF_6"
+      - "EXT_REF_7"
+
+  reference-selection:
+    type: string
+    enum:
+      - "INT_1V2"
+      - "INT_1V8"
+      - "INT_2V4"
+      - "VDD"
+      - "AREF"
+
+  main-mode:
+    type: string
+    enum:
+      - "SE"
+      - "DIFF"
+
+  hysteresis:
+    type: string
+    enum:
+      - "NO_HYST"
+      - "40MV"
+      - "50MV"
+
+  speed-mode:
+    type: string
+    enum:
+      - "LOW"
+      - "NORMAL"
+      - "HIGH"
+
+  threshold-up:
+    type: int
+
+  threshold-down:
+    type: int
+
+  isource:
+    type: string
+    enum:
+      - "OFF"
+      - "IEN_2UA5"
+      - "IEN_5UA"
+      - "IEN_10UA"
```

Comment:
Also these values are quite weird, especially "IEN_2UA5".

Reply:
yeah, but they do match the product spec, so I think it make sense to keep them since they are documented there, and in the device specific header in tree :)

Reply:
Changed the names to just `2UA5` ...